modules/adfBidAdapter.js

```diff
@@ -45,28 +48,58 @@ const NATIVE_PARAMS = {
     name: 'data'
   }
 };
+const OUTSTREAM_RENDERER_URL = 'https://s2.adform.net/banners/scripts/video/outstream/render.js';
 
 export const spec = {
   code: BIDDER_CODE,
   aliases: BIDDER_ALIAS,
   gvlid: GVLID,
-  supportedMediaTypes: [ NATIVE ],
+  supportedMediaTypes: [ NATIVE, BANNER, VIDEO ],
   isBidRequestValid: bid => !!bid.params.mid,
   buildRequests: (validBidRequests, bidderRequest) => {
-    const page = bidderRequest.refererInfo.referer;
+    let app, site;
+
+    const commonFpd = getConfig('ortb2') || {};
+    let { user } = commonFpd;
+
+    if (typeof getConfig('app') === 'object') {
+      app = getConfig('app') || {};
+      if (commonFpd.app) {
+        utils.mergeDeep(app, commonFpd.app);
+      }
+    } else {
+      site = getConfig('site') || {};
+      if (commonFpd.site) {
+        utils.mergeDeep(site, commonFpd.site);
+      }
+
+      if (!site.page) {
+        site.page = bidderRequest.refererInfo.referer;
+      }
+    }
+
+    const device = getConfig('device') || {};
+    device.w = device.w || window.innerWidth;
+    device.h = device.h || window.innerHeight;
+    device.ua = device.ua || navigator.userAgent;
+
     const adxDomain = setOnAny(validBidRequests, 'params.adxDomain') || 'adx.adform.net';
-    const ua = navigator.userAgent;
+
     const pt = setOnAny(validBidRequests, 'params.pt') || setOnAny(validBidRequests, 'params.priceType') || 'net';
-    const tid = validBidRequests[0].transactionId; // ??? check with ssp
+    const tid = validBidRequests[0].transactionId;
     const test = setOnAny(validBidRequests, 'params.test');
-    const publisher = setOnAny(validBidRequests, 'params.publisher');
-    const siteId = setOnAny(validBidRequests, 'params.siteId');
     const currency = config.getConfig('currency.adServerCurrency');
```

Comment:
Just for consistency since you defined getConfig as a const, can you update this line to using that as well instead of calling config.getConfig

Reply:
Updated.